src/gt4py/next/program_processors/runners/dace_iterator/itir_to_tasklet.py

```diff
@@ -909,140 +912,82 @@ def visit_OffsetLiteral(self, node: itir.OffsetLiteral) -> list[ValueExpr]:
-                reduce_tasklet,
-                "__result",
-                result_access,
-                None,
-                dace.Memlet.simple(result_name, "0"),
-            )
         else:
             assert isinstance(node.fun, itir.FunCall)
             assert isinstance(node.fun.args[0], itir.Lambda)
             fun_node = node.fun.args[0]
+            assert isinstance(fun_node.expr, itir.FunCall)
 
-            args = []
-            for node_arg in node.args:
-                if (
-                    isinstance(node_arg, itir.FunCall)
-                    and isinstance(node_arg.fun, itir.SymRef)
-                    and node_arg.fun.id == "neighbors"
-                ):
-                    expr = self.visit(node_arg)
-                    args.append(*expr)
-                else:
-                    args.append(None)
-
-            # first visit only arguments for neighbor selection, all other arguments are none
-            neighbor_args = [arg for arg in args if arg]
-
-            # check that all neighbors expression have the same range
-            assert (
-                len(
-                    set([self.context.body.arrays[expr.value.data].shape for expr in neighbor_args])
-                )
-                == 1
-            )
+            op_name = fun_node.expr.fun
+            assert isinstance(op_name, itir.SymRef)
+            reduce_identity = get_reduce_identity_value(op_name.id, reduce_dtype)
 
-            nreduce = self.context.body.arrays[neighbor_args[0].value.data].shape[0]
-            nreduce_domain = {"__idx": f"0:{nreduce}"}
+            # set reduction state in visit context
+            self.context.reduce_identity = SymbolExpr(reduce_identity, reduce_dtype)
 
-            result_dtype = neighbor_args[0].dtype
-            self.context.body.add_scalar(result_name, result_dtype, transient=True)
+            args = flatten_list(self.visit(node.args))
 
-            assert isinstance(fun_node.expr, itir.FunCall)
-            op_name = fun_node.expr.fun
-            assert isinstance(op_name, itir.SymRef)
+            # clear context
+            self.context.reduce_identity = None
 
-            # initialize the reduction result based on type of operation
-            init_value = get_reduce_identity_value(op_name.id, result_dtype)
-            init_state = self.context.body.add_state_before(self.context.state, "init", True)
-            init_tasklet = init_state.add_tasklet(
-                "init_reduce", {}, {"__out"}, f"__out = {init_value}"
-            )
-            init_state.add_edge(
-                init_tasklet,
-                "__out",
-                init_state.add_access(result_name),
-                None,
-                dace.Memlet.simple(result_name, "0"),
+            # check that all neighbor expressions have the same shape and reduce range
+            nreduce_shape = args[1].value.desc(self.context.body).shape
+            assert len(nreduce_shape) == 1
```

Comment:
Why is this always one?

Reply:
I was thinking that thanks for the conformation, I think we does not need it, but we should maybe add some `if(...) raise NotImplementedError("One one reduction dimension is supported.")` such that we know where to put it.

Reply:
I have actually removed this assertion and the reduction library node now accepts a generic ND-array as input.